### PR TITLE
Ensures flutter is installed

### LINKED ISSUE #207: [Feature Request] Automatic linking
**Is your feature request related to a problem? Please describe.**
I am still running into cases of the wrong SDK being used when switching between branches in my projects that use different Flutter SDK versions. I have the config file committed to VCS and switching a branch also changes the config file.

**Describe the solution you'd like**
I would expect that after a checkout of a different config file and executing an fvm command, I would see either a warning that the wrong SDK is being used or ideally fvm would link and use the correct SDK automatically.

### PR DESCRIPTION
**Pull request by leoafarias**
Closes #207

### CODE REVIEW

**Comment by kuhnroyal** [comment]
I like the improvements they are useful in any case.
But after looking at the code, I think my problem is that `_createSdkLink` is not called in this case. So the SDK link is still the old one which is out of sync with the pinned version.

**Comment by leoafarias** [comment]
@kuhnroyal I understand it. I made further changes to make sure this happens during install

File: packages/cli/lib/src/workflows/install_version.workflow.dart [hunk 1]
```diff
@@ -1,25 +1,38 @@
...
     assert(version != null);
 
     // If it's installed correctly just return and use cached
-    if (await LocalVersionRepo.isInstalled(version)) {
-      FvmLogger.fine('Version: $version - already installed.');
+    final isVersionInstalled = await LocalVersionRepo.isInstalled(version);
+
+    // Ensure the config link and symlink are updated
+    final project = await FlutterProjectRepo.findAncestor();
+    await FlutterProjectRepo.updateSdkLink(project);
```
Comment (kuhnroyal) [inline packages/cli/lib/src/workflows/install_version.workflow.dart:21]: Does this work when the version does not exist?

Comment (kuhnroyal) [reply packages/cli/lib/src/workflows/install_version.workflow.dart:21]: I think this should only be done when the version is pinned. Might need a flag to indicate this. When it is not pinned it should not automatically link as the `use` command is intended to do this.

Comment (leoafarias) [reply packages/cli/lib/src/workflows/install_version.workflow.dart:21]: The previous implementation only did this when the version is pinned. The issue is that the version pinned changed outside of FVM. Essentially we are ensuring that the version in the config always matches the link. 

It checks if the project config does exist to recreate the link, and it goes through the normal install workflow which asks if you want to install the version, or returns if the verison is already installed

**Comment by kuhnroyal** [comment]
I think `install` and `use` should both be able to be called without argument when there is a pinned version.
`install` should ask if it should also `use` the version.
`use` should ask if it should `install` if the version is not found.

**Comment by leoafarias** [comment]
@kuhnroyal I will have to review this, as the use command has some validations to make sure it's running in a project, and install just downloads the version. It seems both commands are now going in the same direction, but want to make sure we can continue to install versions without pinning.

**Pull request merged**